From 660021d371d5698a86e914765239045a669e65fb Mon Sep 17 00:00:00 2001
From: Lennart Poettering <lennart@poettering.net>
Date: Wed, 7 Oct 2015 20:08:49 +0200
Subject: [PATCH 1/2] util: always enforce O_NOCTTY and O_CLOEXEC in
 openpt_in_namespace()

The child process is shortliving, hence always set O_NOCTTY so that the
tty doesn't quickly become controlling TTY and then gives it up again.
Also set O_CLOEXEC, because it's cleaner, and doesn't affect the parent
anyway.
---
 src/basic/util.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/src/basic/util.c b/src/basic/util.c
index 2855993..f845e2d 100644
--- a/src/basic/util.c
+++ b/src/basic/util.c
@@ -6143,7 +6143,7 @@ int openpt_in_namespace(pid_t pid, int flags) {
                 if (r < 0)
                         _exit(EXIT_FAILURE);
 
-                master = posix_openpt(flags);
+                master = posix_openpt(flags|O_NOCTTY|O_CLOEXEC);
                 if (master < 0)
                         _exit(EXIT_FAILURE);
 
-- 
2.6.1

